### ISSUE TITLE
Leading one/zeros and trailing zeros

### ISSUE THREAD

**Issue by oscargus** [apytypes/apytypes#115]
There are situations where one would like to find the actual range of the data (not so much for hardware, but to simulate things), so some sort of leading ones/zeros and possibly trailing zeros method, at least for `APyFixed`, maybe for `APyFixedArray` would be useful.

**Comment by miklhh** [comment]
Agreed. @Theodor-Lindberg is also very interested in this feature. I suggest we do the same thing as is done in ac_types:

> #### `ac_fixed::leading_sign()`
> Returns the number of leading zeros if unsigned. If signed, returns the number of leading 0/1s minus one.

**Comment by oscargus** [comment]
This is implemented for `APyFixed`, but should be implemented for `APyFixedArray` as well.
